arthur/scheduler.py

```diff
@@ -140,6 +140,7 @@ def schedule_task(self, task_id, delay=0):
 
         self._tasks_events[task_id] = event
         task.status = TaskStatus.SCHEDULED
+        self.registry.update(task_id, task)
```

Comment:
Calls to `update` or any other method related to `registry` might cause the code crash. We have to be sure we can handle these errors propertly.